### ISSUE TITLE
source:profile:reconcile throws error when reconciling against an org

### ISSUE THREAD

**Issue by jhawk-kar** [Accenture/sfpowerkit#133]
Running `$ sfdx sfpowerkit:source:profile:reconcile -u targetOrg 'n profileName -d destinationFolder` throws an error:

>EntityDefintion does not support queryMore(), use LIMIT to restrict the results to a single batch

The command works fine if you reconcile against the source only, but against a target org it fails.

sfpowerkit 1.23.1

**Comment by azlam-abdulsalam** [comment]
Thanks for raising the issue,  A  hotfix is released in 1.23.2 which address this temporarily,  as I have explicitly set the version to 46.0 instead of 47.0 which has some breaking changes in Tooling API.  
Please update the plugin to 1.23.2

Will keep this bug open until we have a proper solution

**Comment by jhawk-kar** [comment]
Thanks, @azlam-abdulsalam.  I tried with 1.23.2 and I'm still getting the same error message.  Looks like to readme references `--apversion` as a parameter for this command, but that is not supported, so just a note about that as well.

**Comment by azlam-abdulsalam** [comment]
Can you confirm the plugin version to be 1.23.2 as I am not able to reproduce it?  If you're are using the plugin from the checked out  code, its only merged to release

**Comment by jhawk-kar** [comment]
Running `$ sfdx plugins` lists sfpowerkit as 1.23.2.  I updated from `$ sfdx update`.  Is there a different way I should do the plugin update?  

I also tried with the beta version (1.24.2-1) and I still get the same error.  

My CLI core version is 7.28.7.

**Comment by azlam-abdulsalam** [comment]
To help us debug, can you do the following,
1. Install alpha build with more logging statements, sfdx plugins:install sfpowerkit@alpha
2. Run sfdx sfpowerkit:source:profile:reconcile -u <org> --loglevel=debug  >> output.txt
and send us the debug statment

**Comment by jhawk-kar** [comment]
Debug output
```
Reconciling profile ProfileName.profile-meta.xml
Reconciling App
Application Visiblitilties reduced from 77  to  75
Reconciling Classes
Class Access reduced from 3389  to  3322
Reconciling Fields
```

**Comment by azlam-abdulsalam** [comment]
So it gets stuck at the reconciling fields?  I believe original Entity Definition issue is no longer there..

**Comment by jhawk-kar** [comment]
Not sure. The same error still appears, it just didn't spit out in the debug log. It shows in the terminal window, though.

**Comment by azlam-abdulsalam** [comment]
Thanks, is it a sandbox or scratchorg that you are reconciling?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by jhawk-kar** [comment]
Sandbox.  I'm going to play around with different orgs as well.  Our security model is not the norm, so I want to rule out environment being the problem.

**Comment by jhawk-kar** [comment]
If I run the same query from the `entityDefinitionRetriever.ts` class using the dev console with the `Use Tooling API` box checked, I get the same error.

![image](https://user-images.githubusercontent.com/43937191/67243594-f0a7b200-f425-11e9-8759-fb9bef74163a.png)

Running a `COUNT()` query returns 3250 rows for `EntityDefinition` in our org.

I'm able to run the query with a limit of 2000, but anything more than that throws the error.  So, this is likely due to the large number of rows returned within our environments.  Even filtering out rows returned from managed packages still returns 2102, so filtering out managed package items still wouldn't work in our org.  Let me know your thoughts.

**Comment by azlam-abdulsalam** [comment]
Definitely, it is the problem, and till 47 this query didn't support Limits and Offsets, so I guess we have to change the code to fetch in batches.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by azlam-abdulsalam** [comment]
We will be fixing this on priority. Most probably next week release.

**Comment by azlam-abdulsalam** [comment]
@jhawk-kar  I have done a quick fix with Limits and Offset as with the v47 API. While this  will take a bit of cleanup and testing, would you be able to test the alpha version to validate whether this PoC works well.

Install the review (alpha) build using the command  sfdx plugins:install sfpowerkit@alpha and let me know how it goes

**Comment by jhawk-kar** [comment]
Getting an error running the command with 1.25.2-0
```
Error: Cannot find module '../../../../src/sfpowerkit'
    at Function._load (C:/Users/justin.hawkins/AppData/Local/sfdx/client/7.28.7-7bceba6f24/node_modules/@salesforce/lazy-require/lib/LazyLoader.js:89:24)
    at require (C:/Users/justin.hawkins/AppData/Local/sfdx/client/7.28.7-7bceba6f24/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (C:/Users/justin.hawkins/AppData/Local/sfdx/node_modules/sfpowerkit/lib/impl/metadata/retriever/baseMetadataRetriever.js:3:22)
    at Module._compile (C:/Users/justin.hawkins/AppData/Local/sfdx/client/7.28.7-7bceba6f24/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
```

**Comment by azlam-abdulsalam** [comment]
Need to investigate why it didnt throw a compile error and error in my test runs :(
Fixed in 1.25.3-0,

**Comment by jhawk-kar** [comment]
Thanks, @azlam-abdulsalam.  The reconcile command completed successfully this time.  However, when I went to deploy the profile to the destination that I reconciled against, it failed due several items still being listed in the profile that were not in the destination.  So far, it has complained about:

1. Tab Visibilities
2. App Visibilities
3. Layouts
4. Page Access
5. Class Access

I can open up a separate issue if you feel these aren't related.
